src/metaopt/io/database/mongodb.py

```diff
@@ -0,0 +1,137 @@
+import json
+import bson
+import pymongo
+
+from pymongo import MongoClient
+from datetime import datetime
+from abstractdb import AbstractDB
+
+try:
+    from StringIO import StringIO
+except ImportError:
+    from io import StringIO
+
+try:
+    # Python 3.x
+    from urllib.parse import quote_plus
+except ImportError:
+    # Python 2.x
+    from urllib import quote_plus
+
+
+class MongoDB(AbstractDB):
+    def __init__(self, dbhost='localhost', dbname='metaopt', dbport=27017, \
+                                            username=None, password=None):
+        self.dbhost = str(dbhost)
+        self.dbname = str(dbname)
+        self.username = str(username)
+        self.password = str(password)
+
+        # we might have a different port for the database port
+        # so here we do a check if the port is there
```

Comment:
Is that a TODO comment?

Reply:
I leaved it for a IF block before, I should've removed it. Noted. Thanks